### DISCUSSION TITLE
I'd like to disable requests to __data.json

### DISCUSSION THREAD

**Discussion by Jakeii** [sveltejs/kit#6566] [Ideas]
I'm using [`svelte-query`](https://github.com/SvelteStack/svelte-query) and `+page.server.ts` is where I'm pre-fetching data for the page, and using svelte-query's dehydration/hydration to pass the the data to the client.

Navigating is now ever so slightly slower having to wait for `__data.json` to load on subsequent navigation.

What I'd like is for `+page.server.ts` only be used for the inittal SSR page render. On subsequent page navigation svelte query can handle fetching the data client side as it needs, being able to take advantage of it's cache and have loading states where I need etc.

I've hacked a solution by adding this to the top of the `handle` hook or could look into a web worker to do this.
```js
  if (event.request.url.endsWith('__data.json')) {
    return new Response('{}');
  }
```

But would like to disable the request all together.

**Comment by TranscendentalRide** [comment]
Yes and to add to it, the back button is actually making a call to `__data.js` as well; should be an option to do a full re-render if cache is expired.

**Comment by TranscendentalRide** [comment]
@Jakeii This issue may interest you https://github.com/sveltejs/kit/issues/5699
